Issues when running with Security Manager enabled

When running under JDK 11 with the Security Manager enabled, SmallRye OpenTelemetry experiences an `AccessControlException`. Take this stacktrace from a test from WildFly:

```
Caused by: java.security.AccessControlException: WFSM000001: Permission check failed (permission "("java.lang.RuntimePermission" "getClassLoader")" in code source "(vfs:/content/ContextPropagationTestCase.war/WEB-INF/classes <no signer certificates>)" of "ModuleClassLoader for Module "deployment.ContextPropagationTestCase.war" from Service Module Loader")
  at org.wildfly.security.elytron-base@2.1.0.Final//org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:309)
  at org.wildfly.security.elytron-base@2.1.0.Final//org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:201)
  at java.base/java.lang.ClassLoader.checkClassLoaderPermission(ClassLoader.java:2060)
  at java.base/java.lang.Thread.getContextClassLoader(Thread.java:1487)
  at io.smallrye.opentelemetry@2.3.0//io.smallrye.opentelemetry.implementation.cdi.OpenTelemetryProducer.getOpenTelemetry(OpenTelemetryProducer.java:38)
```

From local testing, it seems that wrapping the code in `OpenTelemetryProducer.getOpenTelemetry()` (and, it turns out, the constructors for `OpenTelemetryClientFilter` and `OpenTelemetryServerFilter`) in a `doPrivileged()` block does the trick, but that kind of change needs to be carefully scrutinized.

Regardless of the final implementation, this failure is causing issues for WildFly and we'd like to help get it fixed.

--- a/implementation/cdi/src/main/java/io/smallrye/opentelemetry/implementation/cdi/OpenTelemetryProducer.java
+++ b/implementation/cdi/src/main/java/io/smallrye/opentelemetry/implementation/cdi/OpenTelemetryProducer.java
@@ -32,10 +32,15 @@ public class OpenTelemetryProducer {
     @Singleton
     public OpenTelemetry getOpenTelemetry() {
         AutoConfiguredOpenTelemetrySdkBuilder builder = AutoConfiguredOpenTelemetrySdk.builder();
+
+        ClassLoader contextClassLoader = SecuritySupport.getContextClassLoader();
+        if (contextClassLoader != null) {
+            builder.setServiceClassLoader(contextClassLoader);
+        }
+
         return builder
                 .setResultAsGlobal(false)
                 .registerShutdownHook(false)
-                .setServiceClassLoader(Thread.currentThread().getContextClassLoader())
                 .addPropertiesSupplier(() -> config.properties())
                 .build()
                 .getOpenTelemetrySdk();

--- a/implementation/cdi/src/main/java/io/smallrye/opentelemetry/implementation/cdi/SecuritySupport.java
+++ b/implementation/cdi/src/main/java/io/smallrye/opentelemetry/implementation/cdi/SecuritySupport.java
@@ -1,0 +1,26 @@
+package io.smallrye.opentelemetry.implementation.cdi;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+class SecuritySupport {
+    private SecuritySupport() {
+        throw new UnsupportedOperationException();
+    }
+
+    static ClassLoader getContextClassLoader() {
+        if (System.getSecurityManager() == null) {
+            return Thread.currentThread().getContextClassLoader();
+        } else {
+            return AccessController.doPrivileged((PrivilegedAction<ClassLoader>) () -> {
+                ClassLoader classLoader = null;
+                try {
+                    classLoader = Thread.currentThread().getContextClassLoader();
+                } catch (SecurityException ex) {
+                    // Ignore
+                }
+                return classLoader;
+            });
+        }
+    }
+}

